app/services/uri_service.rb

```diff
@@ -2,14 +2,17 @@ class URIService
   attr_accessor :uri
 
   def initialize(uri)
+    return nil if uri.nil?
+
     begin 
       @uri = URI.parse(URI.encode(uri))
     rescue URI::InvalidURIError
       nil
     end
   end
 
-  def secure_scheme!
+  def secure_scheme_uri!
     @uri.scheme = "https"
+    @uri
```

Comment:
added this return here to and changed method name to make the interface to the PORO friendlier